From 2ba4b712b5fd061e74c2dfb25afc4cfd9ef7e0ff Mon Sep 17 00:00:00 2001
From: Damien Fouilleul <damien.fouilleul@gmail.com>
Date: Thu, 22 Apr 2021 13:40:58 +0100
Subject: [PATCH] Allow network scan to modify muxes

When 'change muxes' option for network discovery is enabled, allow network scan to modify muxes rather than duplicate them on minor changes such as FEC
---
 src/input/mpegts/mpegts_network_dvb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/input/mpegts/mpegts_network_dvb.c b/src/input/mpegts/mpegts_network_dvb.c
index da85abb087..07c8a1dcd1 100644
--- a/src/input/mpegts/mpegts_network_dvb.c
+++ b/src/input/mpegts/mpegts_network_dvb.c
@@ -703,7 +703,7 @@ dvb_network_create_mux
   }
 
   ln = (dvb_network_t*)mn;
-  mm = dvb_network_find_mux(ln, dmc, onid, tsid, 0, 0);
+  mm = dvb_network_find_mux(ln, dmc, onid, tsid, 0, (ln->mn_autodiscovery == MN_DISCOVERY_CHANGE));
   if (!mm && (ln->mn_autodiscovery != MN_DISCOVERY_DISABLE || force)) {
     save |= dvb_fe_type_by_network_class(ln->mn_id.in_class) == dmc->dmc_fe_type;
     if (save && dmc->dmc_fe_type == DVB_TYPE_S) {

